### ISSUE TITLE
Верно ли, что отладчик и программа должны работать в разных потоках?

### ISSUE THREAD

**Issue by yeputons** [java-course-au/kotlin-course#97]
@dzharkov В списке команд отладчика есть команда `stop`, которая должна останавливать программу. Соответственно, при её вводе у нас должен параллельно исполняться код (если пытаться делать что-то с асинхронностью, получится асинхронный ввод команд, что звучит довольно неприятно). Вывод: код живёт в отдельном потоке.

Если всё так, то какой получается смысл в корутинах конкретно в этом задании? Кажется, что в любом случае надо добавлять какие-то точки останова в код интерпретатора, но вот делать их асинхронными на корутинах или синхронными на conditional variables уже кажется неважным. В целом, "поупражняться в `suspend`" тоже вполне себе цель, конечно.

**Comment by dzharkov** [comment]
@yeputons Нет, команда `stop` нужна для остановки (полного прекращения) программы, которая сейчас заморожена на брейкпойнте. То есть в момент исполнения между брейкпойнтами ввод команд блокируется.

Ну и да, поток должен быть один. И да, это специальное задание, чтобы поэкспериментировать с корутинами
